examples/grf/rhs.jl

```diff
@@ -65,6 +76,56 @@ function make_deterministic_tendency(model::LinearDiffusionSDE{FT, CM, Dirichlet
     return deterministic_tendency!
 end
 
+function make_deterministic_tendency(model::LudoDiffusionSDE{FT, CM}) where {FT, CM}
+    function deterministic_tendency!(du,u,t)
+        N = model.N
+        α = model.α
+        β = model.β
+        γ = model.γ
+
+        # initialize tendency
+        du .= FT(0)
+        
+        # temporary nonlinear vector
+        v = @. tanh(γ*u)
+
+        # apply Laplacian to nonlinear vector
+        for i in 1:N
+            for j in 1:N
+                k = (j-1)*N+i
+                k_ip1 = k+1
+                k_im1 = k-1
+                k_jp1 = k + N
+                k_jm1 = k - N
+                # At the boundary, we have u_bc on the face
+                # while u[k] is at the cell center. -> Δx -> Δx/2
+                if i == N
+                    v_bc = v[(j-1)*N+1]
+                    du[k] += ((v[k_im1] - v[k]) - (v[k] - v_bc)/2)
```

Comment:
I think for periodic BC we do not need the factor of 2. when we supply dirichlet BC, the value is given on the face, so the distance between the center and face is Delta/2. when we use periodic BC, the value is at the adjacent/wrapped cell center, so distance shouldnt be Delta?

Reply:
For reference:
i = j = 1 upper left corner, i = 1, j = N is the upper right corner, i = N, j = 1: lower left corner, i=j=N is the lower right corner

Reply:
Please double check again :)